### PR TITLE
Add `ldexp` tests

### PR DESCRIPTION
**Pull request by zoddicus**
<hr>

**Requirements for PR author:**

- [x] All missing test coverage is tracked with "TODO" or `.unimplemented()`.
- [x] New helpers are `/** documented */` and new helper files are found in `helper_index.txt`.
- [x] Test behaves as expected in a WebGPU implementation. (If not passing, explain above.)

**Requirements for [reviewer sign-off](https://github.com/gpuweb/cts/blob/main/docs/reviews.md):**

- [ ] Tests are properly located in the test tree.
- [ ] [Test descriptions](https://github.com/gpuweb/cts/blob/main/docs/intro/plans.md) allow a reader to "read only the test plans and evaluate coverage completeness", and accurately reflect the test code.
- [ ] Tests provide complete coverage (including validation control cases). **Missing coverage MUST be covered by TODOs.**
- [ ] Helpers and types promote readability and maintainability.

When landing this PR, be sure to make any necessary issue status updates.

### CODE REVIEW

**Comment by zoddicus** [comment]
PTAL

File: src/webgpu/shader/execution/builtin/ldexp.spec.ts [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+      .combine('storageClass', ['uniform', 'storage_r', 'storage_rw'] as const)
+      .combine('vectorize', [undefined, 2, 3, 4] as const)
+  )
+  .fn(async t => {
+    const truthFunc = (e1: number, e2: number): Case | undefined => {
+      const result = e1 * Math.pow(2, e2);
+      // Unclear what the expected behaviour for values that overflow f32 bounds, see https://github.com/gpuweb/gpuweb/issues/2631
+      if (!Number.isFinite(result)) {
+        return undefined;
+      } else if (result > (f32Bits(kBit.f32.positive.max).value as number)) {
```
Comment (ben-clayton) [inline src/webgpu/shader/execution/builtin/ldexp.spec.ts:34]: Can you use your new constants here?

Comment (zoddicus) [reply src/webgpu/shader/execution/builtin/ldexp.spec.ts]: Done, thanks for catching

**Review by ben-clayton — APPROVED** [review]
Good work!

**Pull request merged**